src/v/cloud_storage/remote_partition.cc

```diff
@@ -577,12 +577,10 @@ class partition_record_batch_reader_impl final
                 co_return;
             }
 
-            vlog(
-              _ctxlog.error,
+            throw std::runtime_error(fmt::format(
```

Comment:
>And it actually is at least in one cloud storage time query scenario.

Just making sure I understand this (wording is a little awkward IMO): is it saying that we have seen a scenario where we've incorrectly returned EOF?

Reply:
Updated the wording of the commit. It doesn't mention a specific problem anymore. Will do that in another commit.